src/functions/asyncCredential/sessionService/sessionService.ts

```diff
@@ -74,14 +119,41 @@ export class SessionService implements IRecoverAuthSession {
       result.Items[0].sessionId.S !== ""
     );
   }
+
+  private async checkSessionsExists(authSessionId: string): Promise<boolean> {
```

Comment:
I might be missing something - what's the rationale for this?

If we're checking if there's a recoverable session, is this needed?

Reply:
No, it's for when the random UUID we create is already in the DB. Which is such a tiny posibility. They do it in verifyAuthReq so we followed suit, albeit in a slightly different way.

I spoke to Jason about this and I think if we do include this, I don't think it should throw an error for the user's journey to end, I'd rather just generate a new UUID

Reply:
This approach means that we're always making two requests to create a session.

We _could_ look at doing the PutItemCommand with a Conditional Expression (https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_PutItem.html#API_PutItem_RequestSyntax)

Reply:
Let's continue as is and I'll explore this

Reply:
Will be in next PR